addons/packages/kapp-controller/0.41.2/bundle/config/overlays/update-strategy-overlay.yaml

```diff
@@ -1,6 +1,17 @@
 #@ load("@ytt:overlay", "overlay")
 #@ load("@ytt:data", "data")
 
+#@ def matcher():
+kind: Deployment
+metadata:
+  name: kapp-controller
+spec:
+  template:
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
```

Comment:
Replace the node selector with an affinity rule that can have both please. This would be better than mashing logic into YTT.

See this for an example
https://github.com/vmware-tanzu/community-edition/pull/5416/files#diff-ef1bf4c3aac3352aec9d85c2d4c49ce88853b8f9330240ec981cb8d026d50640L21-L23

Reply:
YTT logic is required since nodeSelector is not hardcoded in the package templates but passed in by the user. Since the user could have defined master nodeSelector we have to do some ytt check to only conditionally replace nodeSelector if master is specified.

We checked k8s 1.21 - 1.24 clusters and all of them have control-plane node selector, so we have decided to replace and not add OR condition.